### PR TITLE
Replaced grpc::string with std::string

### PR DESCRIPTION
**Pull request by veblush**
Previously `grpc::string` was configurable with given custom string type but this indirection added more complexity and this would be better to be replaced with explicit `std::string`. Since we started using `std::string` in gRPC Core, this helps us to have better code readability with just one string representation.

[grpc_build_artifacts_multiplatform](https://sponge.corp.google.com/invocation?id=516cf178-e04a-4b51-ba76-645aa5f1f282&searchFor=): passed

### CODE REVIEW

File: src/compiler/cpp_generator.h [hunk 1]
```diff
@@ -45,86 +45,86 @@ namespace grpc_cpp_generator {
 // Contains all the parameters that are parsed from the command line.
```
Comment (yang-g) [inline src/compiler/cpp_generator.h]: Note you will need to change/remove line 32 to 41.

File: src/compiler/schema_interface.h [hunk 2]
```diff
@@ -41,31 +41,31 @@ namespace grpc_generator {
 // Return formatted comments to be inserted in generated code.
```
Comment (yang-g) [inline src/compiler/schema_interface.h]: line 27-36.

Comment (yang-g) [reply src/compiler/schema_interface.h]: actually I think this is also showing in `src/compiler/config.h` and if we have `std::string` there, this block can just be removed.

File: examples/cpp/metadata/greeter_server.cc [hunk 3]
```diff
@@ -44,7 +44,7 @@ class GreeterServiceImpl final : public Greeter::Service {
 
     // Get the client's initial metadata
     std::cout << "Client metadata: " << std::endl;
-    const std::multimap<grpc::string_ref, grpc::string_ref> metadata = context->client_metadata();
+    const std::multimap<std::string_ref, std::string_ref> metadata = context->client_metadata();
```
Comment (yang-g) [inline examples/cpp/metadata/greeter_server.cc]: I think `string_ref` needs to stay `grpc` :)

Comment (veblush) [reply examples/cpp/metadata/greeter_server.cc]: Oh my bad! I'm fixing it.

File: src/compiler/schema_interface.h (hunk 2 above)
Comment (veblush) [reply src/compiler/schema_interface.h]: You're right. Thanks!

File: src/compiler/cpp_generator.h (hunk 1 above)
Comment (veblush) [reply src/compiler/cpp_generator.h]: You're right. Thanks!

File: include/grpcpp/impl/codegen/config.h [hunk 4]
```diff
@@ -32,9 +32,11 @@
 
 namespace grpc {
 
-typedef GRPC_CUSTOM_STRING string;
-
-using std::to_string;
+// Using std::string and std::to_string is discouraged in favor of
```
Comment (vjpai) [inline include/grpcpp/impl/codegen/config.h]: Change this to using grpc::string and grpc::to_string is discouraged ....

**Pull request closed without merging**